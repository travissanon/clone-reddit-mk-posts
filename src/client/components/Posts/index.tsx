import React from "react";
import Post from "../Post/index";

import "./style.scss";

interface IPostsProps {
	readonly data: IPost[];
}

interface IPost {
	readonly id: string;
	readonly title: string;
	readonly thumbnail: string;
	readonly author: string;
}

const Posts: React.FC<IPostsProps> = ({ data }) => {
	return (
		<div className="reddit-posts">
			<div className="reddit-posts__container">
				{data.map((post: any) => (
					<Post
						key={post.id}
						id={post.id}
						title={post.title}
						thumbnail={post.thumbnail}
						author={post.author}
					/>
				))}
			</div>
		</div>
	);
};

export default Posts;
