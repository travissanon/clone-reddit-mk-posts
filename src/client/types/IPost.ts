export interface IPost {
	readonly id: string;
	readonly title: string;
	readonly thumbnail: string;
	readonly author: string;
	readonly favorite: boolean;
	handleFavoriteClick: (postId: string) => void;
}
