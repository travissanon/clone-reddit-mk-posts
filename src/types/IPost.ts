export default interface IPost {
	readonly id: string;
	readonly created_utc: number;
	readonly title: string;
	readonly thumbnail: string;
	readonly author: string;
}
