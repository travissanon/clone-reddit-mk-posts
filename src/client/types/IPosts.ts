import { IPost } from "./IPost";

export interface IPosts {
	readonly data: IPost[];
	readonly favoritePosts: Set<string>;
	handleFavoriteClick: (postId: string) => void;
}
