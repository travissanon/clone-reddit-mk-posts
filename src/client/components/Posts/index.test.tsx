import * as React from "react";
import * as renderer from "react-test-renderer";

import Posts from "./index";

import { IPosts as IPostsProps } from "@client/types/IPosts";
import { IPost } from "@client/types/IPost";

test("renders content to the page", () => {
	const mockHandleFavoriteClick = (postId: string) => null;
	const postData: IPost[] = [
		{
			id: "test1",
			title: "Test One",
			thumbnail: "",
			author: "tester1",
			favorite: false,
			handleFavoriteClick: mockHandleFavoriteClick,
		},
		{
			id: "test2",
			title: "Test One",
			thumbnail: "",
			author: "tester2",
			favorite: true,
			handleFavoriteClick: mockHandleFavoriteClick,
		},
	];

	const props: IPostsProps = {
		data: postData,
		favoritePosts: new Set(),
		handleFavoriteClick: mockHandleFavoriteClick,
	};

	const tree = renderer.create(<Posts {...props} />).toJSON();
	expect(tree).toMatchSnapshot();
});
