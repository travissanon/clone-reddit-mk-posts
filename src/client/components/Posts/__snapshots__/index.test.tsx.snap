// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content to the page 1`] = `
<div
  className="reddit-posts"
>
  <div
    className="reddit-posts__container"
  >
    <div
      className="reddit-post"
      data-id="test1"
    >
      <div
        className="reddit-post__thumbnail"
      >
        <img
          alt="post thumbnail"
          src=""
        />
      </div>
      <div
        className="reddit-post__content"
      >
        <div
          className="reddit-post__title"
        >
          <h3>
            Test One
          </h3>
        </div>
        <div
          className="reddit-post__author"
        >
          tester1
        </div>
      </div>
      <div
        className="reddit-post__favorite"
        onClick={[Function]}
      >
        <img
          alt="favorite post"
          src={Object {}}
        />
      </div>
    </div>
    <div
      className="reddit-post"
      data-id="test2"
    >
      <div
        className="reddit-post__thumbnail"
      >
        <img
          alt="post thumbnail"
          src=""
        />
      </div>
      <div
        className="reddit-post__content"
      >
        <div
          className="reddit-post__title"
        >
          <h3>
            Test One
          </h3>
        </div>
        <div
          className="reddit-post__author"
        >
          tester2
        </div>
      </div>
      <div
        className="reddit-post__favorite"
        onClick={[Function]}
      >
        <img
          alt="favorite post"
          src={Object {}}
        />
      </div>
    </div>
  </div>
</div>
`;
